### DISCUSSION TITLE
Can composites use "user:" for scoring

### DISCUSSION THREAD

**Discussion by rikmos** [rspamd/rspamd#5134] [Q&A]
I would like to trigger a specific composite based on one account "user:" in the rsd logs , eg: composites.conf  

COMPOSITE_SUBJECT_PHISH {
    expression = "RSPAMD_REDIRECTOR_URL and RSPAMD_FORGED_SENDER_FORWARDING and RSPAMD_SUBJECT_HAS_EXCLAIM";
    user  = baduser
    score = 99.0;
    group = "Multi Language lotto Phishing";
}

Will this work?

**Comment by fatalbanana** [comment]
> Will this work?

No; is it a ChatGPT invention?

Common approach to conditionally disabling symbols is using [conditions in Lua](https://rspamd.com/doc/faq.html#how-can-i-disable-some-rspamd-rules-safely). Unfortunately that won't work for composites but there are other possibilities...

1) Disable the symbol before it is checked:

~~~lua
-- /etc/rspamd/rspamd.local.lua

rspamd_config['DISABLE_SYMBOLS_XXX'] = {
  callback = function(task)
    if task:get_user() ~= 'bob' then
      task:disable_symbol('COMPOSITE_SUBJECT_PHISH')
    end
  end,
}
~~~

2) Add username to symbol options and check that in composites

~~~hcl
# /etc/rspamd/local.d/composites.conf
FOO_COMPOSITE_FOR_BOB {
  expression = "FOO && BAR && OUR_USERNAME[bob]"
}
~~~

~~~lua
rspamd_config['OUR_USERNAME'] = {
  callback = function(task)
    local un = task:get_user()
    if un then return true, un end
}
~~~

3) Use [settings](https://rspamd.com/doc/configuration/settings.html) - you'd want at least two sets of settings - for the user this should be enabled for & everyone else.

**Comment by rikmos** [comment]
I do not want to disable symbols.
I want to know if user can be used to apply a composite. 
Not gpt
